libvast/src/system/archive.cpp

```diff
@@ -71,6 +71,25 @@ archive(archive_type::stateful_pointer<archive_state> self,
                    self->system().render(result.error()));
       return result;
     },
+    [=](stream<event> in) {
+      self->make_sink(
+        in,
+        [](unit_t&) {
+          // nop
+        },
+        [=](unit_t&, const std::vector<event>& xs) {
+          handle_batch(xs);
+        },
+        [=](unit_t&, const error& err) {
+          if (err) {
+            VAST_ERROR(self, "got a stream error:", self->system().render(err));
+          }
+        }
+      );
+    },
+    [=](const std::vector<event>& xs) {
+      handle_batch(xs);
+    },
```

Comment:
Why do we need two overloads for a vector of events?

Reply:
This is for backwards compatibility at the moment. This handler bites the dust once we switch everything over to stream-only processing.

Reply:
Got it, makes sense.